include/pybind11/stl.h

```diff
@@ -144,7 +144,7 @@ template <typename Type, typename Value> struct list_caster {
     using value_conv = make_caster<Value>;
 
     bool load(handle src, bool convert) {
-        if (!isinstance<sequence>(src) || isinstance<str>(src))
+        if (!isinstance<sequence>(src) || isinstance<bytes>(src) || isinstance<str>(src))
```

Comment:
I think we actually don't want this change. Doing so (or well, not doing so) would fix #1807.

I'm OK with making and discussing this change (or undoing your change) in a separate PR, though.

Reply:
See also #2198.